music21/stream/__init__.py

```diff
@@ -6359,6 +6363,32 @@ def makeNotation(self,
             insert=not inPlace,
             # definitely do NOT put a constrainingSpannerBundle constraint
         )
+
+        notesWithSplitDurations: List[note.GeneralNote] = []
+        if splitAtDurations:
+            for container in returnStream.recurse(includeSelf=True, classFilter=('Stream')):
+                for noteObj in container.getElementsByClass('GeneralNote'):
+                    # turn inexpressible durations into complex durations (unless unlinked)
+                    if noteObj.duration.type == 'inexpressible':
+                        noteObj.duration.quarterLength = noteObj.duration.quarterLength
+
+                    # make dotGroups into normal notes
+                    if len(noteObj.duration.dotGroups) > 1:
+                        noteObj.duration.splitDotGroups(inPlace=True)
+                        # TODO: QUESTION(JTW): also need to replace/insert these?
+
+                    # split at durations
+                    if noteObj.duration.type == 'complex':
+                        insertPoint = noteObj.offset
+                        objList = noteObj.splitAtDurations()
+                        container.replace(noteObj, objList[0])
+                        insertPoint += objList[0].quarterLength
+                        for subsequent in objList[1:]:
+                            container.insert(insertPoint, subsequent)
+                            insertPoint += subsequent.quarterLength
+
+                        notesWithSplitDurations += objList
```

Comment:
This should probably be factored out into `def splitAtDurations()`. My next thought is that that exists in `base.py` -- but lacking a way of dealing with stream containers. The only documented uses are on `GeneralNote`, but it accepts anything:

```
>>> s = stream.Stream()
>>> n = note.Note(quarterLength=5)
>>> s.insert(0, n)
>>> s.duration
<music21.duration.Duration 5.0>
>>> s.splitAtDurations()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jwalls/music21/music21/base.py", line 3160, in splitAtDurations
    splitList = self.splitByQuarterLengths(quarterLengthList)
  File "/Users/jwalls/music21/music21/base.py", line 3041, in splitByQuarterLengths
    spannerList.extend(st.spannerList)
AttributeError: 'tuple' object has no attribute 'spannerList'
```

So maybe what is sketched out here (cribbed from the existing xml exporter) should move to `base.splitAtDurations`, wrapping what is already there?

Reply:
Well, no, it should probably just be on stream.Stream so that it overrides base.splitAtDurations for Stream subclasses only.